### ISSUE TITLE
`async with` doesn't allow newlines

### ISSUE THREAD

**Issue by hrmorley34** [ipython/ipython#12975]
When writing a `with` statement, pressing enter goes onto a new line, rather than immediately running the code:
```py
In [26]: with example:
    ...:     do_1()[enter]
    ...:     [<-pointer goes here]
```
However, when using `async with`, this does not apply to any lines except for the first after the colon:
```py
In [26]: async with example:
    ...:     do_1()[enter]
[code is run]
```

(Tested in latest PyPI: `IPython.__version__ == "7.23.1"`)

**Comment by MrMino** [comment]
Indeed, it's still the case on master (`8.0.0.dev`). Thanks for reporting @hrmorley34.

**Comment by Carreau** [comment]
this should be handled in IPython/terminal/shortcuts.py:newline_or_execute_outer  which itself will rely on IPython/core/inputtransformer2.py:check_complete, this is likely where the logic should be tested as to why it eecutes and does not insert a new line.

**Comment by madbird1304** [comment]
PR: https://github.com/ipython/ipython/pull/13090

**Comment by hrmorley34** [comment]
I've just come back to this issue, and it doesn't seem to be fixed; I've tested with IPython 7.31.0 in Python 3.8.10, 3.9.9, and 3.10.1, with IPython 7.28.0 (when the backport was introduced) in Python 3.10.1, and with IPython 8.0.0.dev (`pip install git+https://github.com/ipython/ipython`) in Python 3.10.1; all of these versions have the same problem as in the original example.

**Comment by Carreau** [comment]
Yes you are right, test were not properly added in previous fix, I just made #13436 that should fix it.
